File: frontend/src/types/assessment.ts
export const assessmentDomains = ['visual-symbol-discrimination', 'orthographic-recognition', 'phonological-awareness', 'working-memory', 'sequencing', 'attention-visual-search', 'reading-fluency', 'interaction-behavior'] as const;
export type AssessmentDomain = typeof assessmentDomains[number];

export type AssessmentEventType = 'trial_started' | 'stimulus_presented' | 'click' | 'selection' | 'response' | 'correct_response' | 'incorrect_response' | 'trial_completed' | 'gaze_sample' | 'fixation_started' | 'fixation_ended';

export interface TrialRecord { id: string; trialNumber: number; stimulus: string; expectedResponse?: string; actualResponse?: string; correctness?: boolean; reactionTimeMs?: number; hesitationTimeMs?: number; }
export interface InteractionEvent { type: AssessmentEventType; timestamp: string; payload: Record<string, unknown>; }

File: backend/app/assessment.py
from enum import StrEnum


class AssessmentDomain(StrEnum):
    VISUAL_SYMBOL_DISCRIMINATION = "visual-symbol-discrimination"
    ORTHOGRAPHIC_RECOGNITION = "orthographic-recognition"
    PHONOLOGICAL_AWARENESS = "phonological-awareness"
    WORKING_MEMORY = "working-memory"
    SEQUENCING = "sequencing"
    ATTENTION_VISUAL_SEARCH = "attention-visual-search"
    READING_FLUENCY = "reading-fluency"
    INTERACTION_BEHAVIOR = "interaction-behavior"


RESEARCH_FEATURE_NAMES = (
    "mean_fixation_duration",
    "max_fixation_duration",
    "fixation_count",
    "target_fixation_duration",
    "distractor_fixation_duration",
    "distractor_target_ratio",
    "ttff",
    "fixation_switches",
    "reaction_time",
    "hesitation_time",
    "saccade_variance",
    "regression_count",
    "scanpath_entropy",
    "horizontal_saccade_bias",
)
